### DISCUSSION TITLE
strongly-typed-ids-in-ef-core-using-strongly-typed-entity-ids-to-avoid-primitive-obsession-part-4/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#20] [Announcements]
# strongly-typed-ids-in-ef-core-using-strongly-typed-entity-ids-to-avoid-primitive-obsession-part-4/

In this post I describe how to use strongly-typed IDs with EF Core by using value converters and a custom ValueConverterSelector, which actually works!

https://andrewlock.net/strongly-typed-ids-in-ef-core-using-strongly-typed-entity-ids-to-avoid-primitive-obsession-part-4/

**Comment by WilliamRees** [comment]
Great post it saved me a lot of time! Do you plan on adding the StronglyTypedIdValueConverterSelector to one of the nuget packages?

**Reply by WilliamRees** [reply]
I also tried to use the StronglyTypedIdVauleConverterSelector with ef migration and it seemed like the replaced value converter selector wasn't working. Have you tried this with ef core migrations and would you have any suggestions?

**Comment by panayot-zhi** [comment]
v1.0.0-beta06 of StronglyTypedId generates NestedClass named EfCoreValueConverter, not StronglyTypedIdEfValueConverter

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at April 24 2019, 10:24, in reply to [Steve Hansen](https://disqus.com/by/xiugraag/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/20#discussioncomment-6056487)</em>

---

<p>You should be able to use the same approach I have, derive from my class, call the `base` method, and add your own implementation.</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at May 24 2019, 10:53, in reply to [Peter](https://disqus.com/by/disqus_TbyTNSxzwn/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/20#discussioncomment-6056497)</em>

---

<p>That's not something I've looked into I'm afraid. I don't use EF Core a huge amount, and I haven't tried it out with the strongly typed IDs. I'm a little surprised it didn't work though, I'll try digging in and see what's going on at some point soon</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at May 29 2019, 08:46, in reply to [Kamil Grzybek](https://disqus.com/by/kamil_grzybek/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/20#discussioncomment-6056499)</em>

---

<p>Glad to see you had some success :) And thanks for pointing out the typo!</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at May 29 2019, 08:49, in reply to [Kori](https://disqus.com/by/djbyter/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/20#discussioncomment-6056503)</em>

---

<p>Unfortunately the whole approach with EfCore feels like a hack :( Thanks for the feedback. The other place this doesn't work it seems is if you're using database generated IDs</p>

[a comment by an automated account is not shown]
